src/KeyManagement.sol

```diff
@@ -33,6 +33,7 @@ abstract contract KeyManagement is IKeyManagement {
 
     function update(bytes32 keyHash, Settings settings) external {
         _onlyThis();
+        if (keyHash == KeyLib.ROOT_KEY_HASH) revert CannotUpdateRootKey();
```

Comment:
I think we also have to check in register that you don't register the root key public key or something, because you could register it, and then it's `key.hash()` would return not bytes32(0), so you could techinically update it lol

Reply:
also this means we can't add hooks to root key